tests/trainer/test_sharded_checkpoint.py

```diff
@@ -293,53 +346,179 @@ def test_fsdp_full_state_dict_load_with_ema(world_size, tmp_path: pathlib.Path,
+
+
+@pytest.mark.gpu
+@pytest.mark.remote
+@world_size(2)
+@pytest.mark.parametrize('state_dict_type', ['sharded'])
+@pytest.mark.parametrize('precision', ['amp_bf16', 'amp_fp16'])
+@pytest.mark.parametrize('autoresume', [False])  #, True])  # True commented out for now
+@pytest.mark.parametrize('num_shards', [2, 4, 7])
+@pytest.mark.parametrize('sharding_strategy', ['FULL_SHARD'])
+@pytest.mark.skipif(version.parse(torch.__version__) < version.parse('2.0.1'), reason='requires PyTorch 2.01 or higher')
+@pytest.mark.filterwarnings(r'ignore:TypedStorage is deprecated.:UserWarning')
+def test_elastic_resumption(world_size, tmp_path: pathlib.Path, state_dict_type: str, autoresume: bool, precision: str,
+                            sharding_strategy, s3_bucket, s3_read_only_prefix, num_shards: int):
+    if state_dict_type == 'local' and using_torch_2():
+        pytest.xfail(
+            'Loading a state_dict_type="local" checkpoint with strict=True errors out. See https://github.com/pytorch/pytorch/issues/102667 for more info'
+        )
+    if autoresume:
+        run_name = 'my-autoresume-run'
+    else:
+        run_name = None
+
+    base_path = f's3://{s3_bucket}/{s3_read_only_prefix}/elastic_test/{sharding_strategy.lower()}_{state_dict_type}_{precision}_{num_shards}/'
+
+    mono_load_path = os.path.join(base_path, 'mono.pt')
+    mono_trainer = get_trainer(
+        fsdp_state_dict_type='full',
+        load_path=mono_load_path,
+        num_features=32,  # This parameter setting is very important. Don't change or the test will fail.
+        num_classes=8,  # This parameter setting is very important. Don't change or the test will fail.
         precision=precision,
-        autoresume=autoresume,
+        autoresume=False,  # Hardcoded to false b/c mono checkpoints don't have symlinks to them.
         run_name=run_name,
         max_duration='4ba',
-        optimizer=optimizer,
     )
-    state_dict_from_trainer2 = trainer2.state.state_dict()
+    if autoresume:
+        sharded_load_path = None
+        save_folder = base_path
+    else:
+        save_folder = None
+        sharded_load_path = os.path.join(base_path, 'ba2')
+        assert not is_checkpoint_legacy_sharded(object_store=S3ObjectStore(bucket=f'{s3_bucket}'),
+                                                source_path=sharded_load_path.replace(f's3://{s3_bucket}/', ''))
 
-    # Compare saved state and loaded state for both ranks.
-    _compare_model_params_between_state_dicts(state_dict_from_trainer1, state_dict_from_trainer2)
-    _compare_optims_between_state_dicts(state_dict_from_trainer1, state_dict_from_trainer2)
-    _compare_metrics_between_state_dicts(state_dict_from_trainer1, state_dict_from_trainer2)
+    sharded_trainer = get_trainer(
+        fsdp_state_dict_type=state_dict_type,
+        save_folder=save_folder,
+        load_path=sharded_load_path,
+        num_features=32,  # This parameter setting is very important. Don't change or the test will fail.
```

Comment:
can you say why its important (cause has to match the saved checkpoint)

Reply:
added clarifying comment